nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/ProcessorController.java

```diff
@@ -878,47 +962,73 @@ public String getCurrentScript() {
         return textArea.getText();
     }
 
-    synchronized void setProcessingOn() {
-        isProcessing = true;
-        doProcessWhenDone = false;
-    }
-
-    synchronized void setProcessingOff() {
-        isProcessing = false;
+    void finishProcessing(Dataset dataset) {
+        ConsoleUtil.runOnFxThread(() -> finishOnPlatform(dataset));
     }
 
-    boolean isProcessing() {
-        return isProcessing;
+    void finishOnPlatform(Dataset dataset) {
+        String oldName = dataset.getFile().toString();
+        String newName = oldName;
```

Comment:
These two lines aren't used, not sure if they should be removed or used someplace?